### PR TITLE
#26857 Cretae API and Factory method to get Unique Workflow SubActions

### LINKED ISSUE #26857: Metrics: Count of unique workflow subactions in all Workflow actions
### Parent Issue

https://github.com/dotCMS/core/issues/25889

### User Story

As a user of the dotCMS backend, I want to be able to Count of unique workflow subactions in all Workflow actions

### Acceptance Criteria

Have an API method to Count of unique workflow subactions in all Workflow actions

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by freddyDOTCMS**
### Proposed Changes
* Create API and Factory method to get Unique Workflow SubActions

### CODE REVIEW

**Review by victoralfaro-dotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_2.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * WorkFlowActionlet use on {@link WorkflowFactoryTest#countAllWorkflowUniqueSubActions()}
+ * NOT USE IT ON ANOTHER TEST
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_2.java:14]: DO NOT*

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_4.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * WorkFlowActionlet use on {@link WorkflowFactoryTest#notCountUniqueSubActionsFromArchivedSchema()}
+ * NOT USE IT ON ANOTHER TEST
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_4.java:14]: DO NOT*

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_5.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * WorkFlowActionlet use on {@link WorkflowAPITest#countAllWorkflowUniqueSubActions()} ()}
+ * NOT USE IT ON ANOTHER TEST
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_5.java:14]: DO NOT

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPI.java [hunk 4]
```diff
@@ -1115,6 +1115,13 @@ SystemActionWorkflowActionMapping mapSystemActionToWorkflowActionForWorkflowSche
 	 */
 	long countAllSchemasSubActions(User user) throws DotDataException, DotSecurityException;
 
+	/**
+	 * Return the count of Unique SubAction in all Action
+	 *
+	 * @return
+	 */
+	long countAllSchemasUniqueSubActions(User user) throws DotDataException, DotSecurityException;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPI.java:1123]: Change to: Returns the count of unique subactions in all Workflow actions

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_2.java (hunk 1 above)
Comment (freddyDOTCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_2.java]: done https://github.com/dotCMS/core/pull/26869/commits/e997967ea222c2992306f6c3c25dc24d2c006b53#diff-84a90622d17c3a96bb15a5443ddcc5a16dd4778fad031f88fd4777866cb3486aR14

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_3.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * WorkFlowActionlet use on {@link WorkflowFactoryTest#countAllWorkflowUniqueSubActions()}
+ * NOT USE IT ON ANOTHER TEST
```
[the remark this answers is not in the document]
Comment (freddyDOTCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_3.java]: done https://github.com/dotCMS/core/pull/26869/commits/e997967ea222c2992306f6c3c25dc24d2c006b53#diff-84a90622d17c3a96bb15a5443ddcc5a16dd4778fad031f88fd4777866cb3486aR14

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_4.java (hunk 2 above)
Comment (freddyDOTCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/business/Actionlet_4.java]: done https://github.com/dotCMS/core/pull/26869/commits/e997967ea222c2992306f6c3c25dc24d2c006b53#diff-26a16a2d3752b681d20e21a3db497e40e98a7d588bd24d6104ce1b3c85b3d55dR14

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkFlowFactory.java [hunk 6]
```diff
@@ -509,4 +509,13 @@ public void saveSchemesForStruct(String contentTypeInode, List<WorkflowScheme> s
 	 * @return
 	 */
 	long countAllSchemasSubActions() throws DotDataException;
+
+
+	/**
+	 * Return the count of Unique SubAction in all Action
```
[the remark this answers is not in the document]
Comment (freddyDOTCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkFlowFactory.java]: done https://github.com/dotCMS/core/pull/26869/commits/e997967ea222c2992306f6c3c25dc24d2c006b53#diff-9cbe979f949605cb72c988900af572a353ba138a6ddb53f257bb8f04e98d5606R515

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPI.java (hunk 4 above)
Comment (freddyDOTCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPI.java:1123]: done https://github.com/dotCMS/core/pull/26869/commits/e997967ea222c2992306f6c3c25dc24d2c006b53#diff-d7c7d1cb2ec9c3b8af52d37d58c0b0ac6705f249676a0f454b554c2690311864R1119

**Review by dsilvam — APPROVED** [review]

**Pull request merged**